### DISCUSSION TITLE
i have a large list like this ['z','z','z','z','e','e','e','z','z']
i want to print '4z3e2z'
help please

### DISCUSSION THREAD

**Discussion by vkp3803** [Tanu-N-Prabhu/Python#31] [Q&A] [answered]
i have a large list like this ['z','z','z','z','e','e','e','z','z']
i want to print '4z3e2z'
help please

**Comment by MOHAMAD101991** [comment]
def count_consecutive_chars(lst):
    result = ""
    count = 1
    for i in range(1, len(lst)):
        if lst[i] == lst[i-1]:
            count += 1
        else:
            result += str(count) + lst[i-1]
            count = 1
    result += str(count) + lst[-1]
    return result

lst = ['z','z','z','z','e','e','e','z','z']
print(count_consecutive_chars(lst))

**Comment by TatyOko28** [comment]
To solve this, you need to group consecutive identical elements in the list and count how many times each element appears consecutively. Here's how you can do it:
```bash
from itertools import groupby

# Input list
data = ['z', 'z', 'z', 'z', 'e', 'e', 'e', 'z', 'z']

# Group consecutive identical elements and count occurrences
result = ''.join(f"{len(list(group))}{key}" for key, group in groupby(data))

# Print the result
print(result)

```
Output:
```bash
4z3e2z

```

**Comment by Tanu-N-Prabhu** [comment, marked as answer]
Try this ;)

```
def compress_list(data):
    if not data:
        return ""

    result = []
    count = 1

    for i in range(1, len(data)):
        if data[i] == data[i - 1]:
            count += 1
        else:
            result.append(f"{count}{data[i - 1]}")
            count = 1
    # Append the last group
    result.append(f"{count}{data[-1]}")
    
    return ''.join(result)

# Example usage
data = ['z','z','z','z','e','e','e','z','z']
print(compress_list(data))  # Output: 4z3e2z
```

## Explanation:
* Loop through the list and keep a count of how many times the current character repeats.
* When the current character changes, store the count and character in the result, and reset the count.
* Finally, return the joined string.
